backend/src/middlewares/permission/permission.middleware.ts

```diff
@@ -0,0 +1,36 @@
+import { FastifyReply, FastifyRequest } from 'fastify';
+
+import { HttpCode, Permissions } from '~/common/enums/enums';
+import { PermissionError } from '~/exceptions/exceptions';
+import { groupsToPermissions, usersToGroups } from '~/services/services';
+
+const hasPermission = (permissionId: Permissions) => {
+  return async (req: FastifyRequest, reply: FastifyReply): Promise<void> => {
+    try {
+      const allUserGroups = await usersToGroups.getGroupsByUserId(req.user.id);
+      const groupsIds = allUserGroups.items?.map((group) => group.groupId);
+
+      if (!groupsIds) {
+        throw new PermissionError();
+      }
+
+      const permissions = [];
+      for (const groupId of groupsIds) {
+        const groupPermissions =
+          await groupsToPermissions.getPermissionsByGroupId(groupId);
+        const permissionsIds = groupPermissions.map(
+          (item) => item.permissionId,
+        );
+        permissions.push(...permissionsIds);
+      }
```

Comment:
As we already have `user` entity injected to request it could be much more easy to get user with permissions on `authorization.plugin` and then use that user-with-permissions entity to check if any actions are allowed - then you won't need to have neither `usersToGroups.getGroupsByUserId` nor `groupsToPermissions.getPermissionsByGroupId` to be called in middleware

Reply:
I just saw that Oleg has a task "Return user by sign in sign up with a permission array", so as I understand, I dont need all this logic with getting user permissions, only req.user.permissions.

But it`s not merged in develop yet

Reply:
@andanfe